src/util/StreamUtil.ts

```diff
@@ -23,6 +23,12 @@ export async function readableToString(stream: Readable): Promise<string> {
   return (await arrayifyStream(stream)).join('');
 }
 
+// These error messages usually indicate expected behaviour so should not give a warning.
+const safeErrors = new Set([
+  'Cannot call write after a stream was destroyed',
+  'premature close',
+]);
```

Comment:
I think we should commit to `.errorCode` instead and use `ERR_STREAM_DESTROYED` and `ERR_STREAM_PREMATURE_CLOSE`.

Reply:
Where would we use this `errorCode`? The errors we're receiving have been thrown by Node.js, the callback here is the first time we see these errors in our scope.

Reply:
This code is set by Node.js and a more durable way of identifying its native errors than error messages, which might change over time.

Reply:
Learn something new every day.

Did some testing on the support for this on older node versions. This is how I discovered that `premature close` is the one being thrown on node 14, the other one is 16. That second one has a code, but the first one is thrown here: https://github.com/mafintosh/end-of-stream/blob/daba5d692f7f016bad7831b4f61caad3ba2d2544/index.js#L54 (this is a dependency of the pump library we use).

Probably caused by changes in stream behaviour but the problem is that the second one only has that string.

We could check the code and then go to a list like this as backup in case it's undefined.

Reply:
Good as is then, thanks for checking. Perhaps add a comment as to why we're not checking error code (we do this for `ENOENT`, for instance).